forecasting/src/autogluon/forecasting/models/gluonts_model/auto_tabular/auto_tabular_model.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+from autogluon.core.utils.exceptions import TimeLimitExceeded
+from gluonts.nursery.autogluon_tabular.estimator import TabularEstimator
+
+from ..abstract_gluonts import AbstractGluonTSModel
+
+logger = logging.getLogger(__name__)
+
+
+class AutoTabularModel(AbstractGluonTSModel):
+    """
+    Autotabular model from Gluon-TS
+    """
+    def __init__(self, path: str, freq: str, prediction_length: int, name: str = "AutoTabular",
+                 eval_metric: str = None, hyperparameters=None, model=None, **kwargs):
+        super().__init__(path=path,
+                         freq=freq,
+                         prediction_length=prediction_length,
+                         hyperparameters=hyperparameters,
+                         name=name,
+                         eval_metric=eval_metric,
+                         model=model,
+                         **kwargs)
+
+    def create_model(self):
+        if len(AbstractGluonTSModel.prev_fitting_time) == 0:
+            time_limit = 300
+        else:
+            time_limit = max(sum(AbstractGluonTSModel.prev_fitting_time) / len(AbstractGluonTSModel.prev_fitting_time), 300)
+        if "time_limit" not in self.params:
+            self.params["time_limit"] = time_limit
+        self.model = TabularEstimator(freq=self.params["freq"],
+                                      prediction_length=self.params["prediction_length"],
+                                      time_limit=self.params["time_limit"],
+                                      last_k_for_val=self.params["prediction_length"])
+
+    def fit(self, train_data, val_data=None, time_limit=None):
+        if time_limit is None or time_limit > 0:
```

Comment:
> Maybe I missed something, this `if` will always be true unless `time_limit` can be negative.

Maybe you can take a look at the code [here](https://github.com/awslabs/autogluon/pull/1538/files/2e62e728d4e1efcc574c7e1e2a1e51b3387d9f6f#diff-3a6597115bab6b198283037deb8e5edef4fc7ce7ab009d6c91ccafe828855554R51), `time_limit` is reduced to `time_left` each time a model is trained and eventually it will be reduced to less than 0 and that's when `time_limit `is used up and the process will stop.

Reply:
> I see, thanks. Is this logic defined at line 54 in model_trial.py? If not, please point me to the line. If yes, when `time_left <= 0`, it will raise an exception so `time_left > 0` will always be true right?

I see what you mean and you seem to be correct. The time logic here is indeed a bit confusing and might need some improvements.

Reply:
I agree there seems to be a larger issue here. The superclass seems to keep track of "previous fitting times" in a class variable, which should not be the case seeing especially it is an abstract class. I will add todos.